pkg/inject/sidecar.go

```diff
@@ -485,3 +513,21 @@ func isContainerPortAvailable(port int32, dep *appsv1.Deployment) bool {
 	}
 	return true
 }
+
+// GetEnvFromConfigMaps return configMap which matches with configMapRef
+func GetConfigMapsMatchedEnvFromInDeplyment(dep appsv1.Deployment, configMaps []corev1.ConfigMap) []corev1.ConfigMap {
+	configMapSearchMap := make(map[string]corev1.ConfigMap)
+	for _, cm := range configMaps {
+		configMapSearchMap[cm.Name] = cm
+	}
```

Comment:
I'm not sure if it is a good idea to copy the ConfigMaps in a slice just to do the search. Can we avoid that?

Reply:
"Thank you for your feedback! "
Instead of iterating over the entire configMaps slice for each envConfigMap.ConfigMapRef, I've created a map where the keys are the names of the corev1.ConfigMap objects and the values are the corev1.ConfigMap objects themselves.
With this map in place, we can directly find the matching corev1.ConfigMap for a given envConfigMap.ConfigMapRef.Name without the need for multiple iterations over the configMaps slice. This optimization reduces the time complexity of finding a matching corev1.ConfigMap